### ISSUE TITLE
Can not implement Combining Population With Pagination

### ISSUE THREAD

**Issue by wemod123** [strapi/strapi#12540]
## Can not implement Combining Population With Pagination

Following [this document](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest/populating-fields.html#combining-population-with-other-operators), to implement **pagination** in **population**, it should work, for example:

```
const qs = require('qs');
const query = qs.stringify({
  fields: ['title', 'slug'],
  populate: {
    comments: {
      fields: ['name'],
      pagination: {
          start: 50,
          limit: 200
      }
    },
  },
}, {
  encodeValuesOnly: true,
});

await request(`/api/articles?${query}`);
```

But it not work as expected

### Steps to reproduce the behavior

1.  As described above, implement pagination in API call
2.  API response not respect to pagination in population

### Expected behavior

If the document is correct, pagination in population should work.

If this is not designed to be work in this way, the document should be modified to reflect what had been designed.

### System

- Node.js version: v14
- Strapi version: v4.0.7"
- Database: pg
- Operating system: windows

**Comment by derrickmehaffy** [comment]
@alexandrebodin correct me if I'm wrong but we can't do this due the performance hit on how we generate joins right?

**Comment by alexandrebodin** [comment]
Yes the doc is wrong this is not supported. Adding pagination would result in running one query per entry to get accurate pagination.

**Comment by derrickmehaffy** [comment]
I'll transfer this over to the documentation repo then to be fixed as a bug. @wemod123 we will not be able to add this as from a technical perspective it's simply not possible.
